prow/gerrit/client/client.go

```diff
@@ -507,12 +516,21 @@ func (c *Client) GetBranchRevision(instance, project, branch string) (string, er
 
 // Account returns gerrit account for the given instance
 func (c *Client) Account(instance string) (*gerrit.AccountInfo, error) {
+	c.lock.RLock()
+	existing, ok := c.accounts[instance]
+	c.lock.RUnlock()
+	if ok {
+		return existing, nil
+	}
+
+	// Looks like we need to populate the value so get the write lock, but then check again.
 	c.lock.Lock()
 	defer c.lock.Unlock()
 	if existing, ok := c.accounts[instance]; ok {
```

Comment:
I guess we only do this check because we relinquished the lock in line 521, and another thread could have populated the value for us by the time we re-lock in line 527 above?

Reply:
Yes exactly. This check prevents redundant API calls in that case.